src/azure-cli/azure/cli/command_modules/cdn/commands.py

```diff
@@ -90,6 +91,30 @@ def _inner_not_found(ex):
                                  custom_func_name='update_endpoint',
                                  doc_string_source='azure.mgmt.cdn.models#EndpointUpdateParameters',
                                  supports_no_wait=True)
+        g.generic_update_command('add-rule', setter_name='update', setter_arg_name='endpoint_update_properties',
+                                 custom_func_name='add_rule',
+                                 doc_string_source='azure.mgmt.cdn.models#EndpointUpdateParameters',
+                                 supports_no_wait=True)
+        g.generic_update_command('remove-rule', setter_name='update', setter_arg_name='endpoint_update_properties',
+                                 custom_func_name='remove_rule',
+                                 doc_string_source='azure.mgmt.cdn.models#EndpointUpdateParameters',
+                                 supports_no_wait=True)
+        g.generic_update_command('add-condition', setter_name='update', setter_arg_name='endpoint_update_properties',
+                                 custom_func_name='add_condition',
+                                 doc_string_source='azure.mgmt.cdn.models#EndpointUpdateParameters',
+                                 supports_no_wait=True)
+        g.generic_update_command('remove-condition', setter_name='update', setter_arg_name='endpoint_update_properties',
+                                 custom_func_name='remove_condition',
+                                 doc_string_source='azure.mgmt.cdn.models#EndpointUpdateParameters',
+                                 supports_no_wait=True)
+        g.generic_update_command('add-action', setter_name='update', setter_arg_name='endpoint_update_properties',
+                                 custom_func_name='add_action',
+                                 doc_string_source='azure.mgmt.cdn.models#EndpointUpdateParameters',
+                                 supports_no_wait=True)
+        g.generic_update_command('remove-action', setter_name='update', setter_arg_name='endpoint_update_properties',
+                                 custom_func_name='remove_action',
+                                 doc_string_source='azure.mgmt.cdn.models#EndpointUpdateParameters',
+                                 supports_no_wait=True)
 
```

Comment:
[Command Guidline](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md#command-naming-and-behavior) 

According to your code, command should force to design like this:
```
az cdn rule add/list/remove
az cdn condition add/list/remove
az cdn action add/list/remove
```